lightning/src/routing/router.rs

```diff
@@ -866,9 +869,9 @@ where L::Target: Logger {
 					// In order to already account for some of the privacy enhancing random CLTV
 					// expiry delta offset we add on top later, we subtract a rough estimate
 					// (2*MEDIAN_HOP_CLTV_EXPIRY_DELTA) here.
-					let max_total_cltv_expiry_delta = payment_params.max_total_cltv_expiry_delta
+					let max_total_cltv_expiry_delta = (payment_params.max_total_cltv_expiry_delta - final_cltv_expiry_delta)
 						.checked_sub(2*MEDIAN_HOP_CLTV_EXPIRY_DELTA)
-						.unwrap_or(payment_params.max_total_cltv_expiry_delta);
+						.unwrap_or(payment_params.max_total_cltv_expiry_delta - final_cltv_expiry_delta);
```

Comment:
To help clarify my understanding, does `$next_hops_cltv_delta` include `final_cltv_expiry_delta`? Or is this needed because `$candidate.cltv_expiry_delta()` is actually for the previous hop as adjusted later on line 1534 and thus `final_cltv_expiry_delta ` was not accounted for?

Reply:
If I understand correctly, the latter is the case: at this point the `final_cltv_expiry_delta` has not been added and therefore needs to be accounted for. This is in contrast to later usages of `payment_params.max_total_cltv_expiry_delta`, as for example in `add_random_cltv_offset()`, where `final_cltv_expiry_delta` is already part of the paths' CTLV deltas.